pages/category/[id].tsx

```diff
@@ -75,11 +75,20 @@ const Category = ({ posts, category, error }: Props) => {
   </>
 };
 
-export const getServerSideProps = async ({ query }) => {
+export const getStaticPaths = async () => {
+  const apiRef = new API();
+  const categories = await apiRef.getCategories();
```

Comment:
Expose a new method `getCategoriesPaths` or similar, that only fetches the slugs to make the request lighter. Also check how contentful can handle this.